### PR TITLE
APS-1746 add premise site survey import seed job

### PR DESCRIPTION
**Pull request by davidatkinsuk**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/seed/cas1/Cas1SeedPremisesFromSiteSurveyXlsxJob.kt [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+      point = premisesInfo.point
+      gender = siteSurvey.maleFemale.toApprovedPremisesGender()
+    }
+
+    existingPremise.characteristics.clear()
+    existingPremise.characteristics.addAll(premisesInfo.characteristicEntities)
+
+    val afterChange = ApprovedPremisesForComparison.fromEntity(existingPremise)
+
+    val diff = javers.compare(beforeChange, afterChange)
```
Comment (danhumphreys-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/seed/cas1/Cas1SeedPremisesFromSiteSurveyXlsxJob.kt:193]: Nice - looks like a useful library

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/seed/cas1/Cas1SeedPremisesFromSiteSurveyXlsxJob.kt [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+        status = entity.status,
+        point = entity.point,
+        gender = entity.gender,
+        supportsSpaceBookings = entity.supportsSpaceBookings,
+        managerDetails = entity.managerDetails,
+      )
+    }
+  }
+
+  @SuppressWarnings("TooGenericExceptionThrown")
```
Comment (danhumphreys-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/seed/cas1/Cas1SeedPremisesFromSiteSurveyXlsxJob.kt:261]: There is a new `SiteSurveyImportException` class which might be applicable here.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/seed/cas1/Cas1SiteSurveyPremiseFactory.kt [hunk 3]
```diff
@@ -0,0 +1,160 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.seed.cas1
+
+import org.jetbrains.kotlinx.dataframe.AnyCol
+import org.jetbrains.kotlinx.dataframe.DataFrame
+import org.jetbrains.kotlinx.dataframe.api.getColumn
+import org.jetbrains.kotlinx.dataframe.name
+
+class Cas1SiteSurveyPremiseFactory {
```
Comment (danhumphreys-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/seed/cas1/Cas1SiteSurveyPremiseFactory.kt:8]: We should be using factories for the room characteristic job too..

**Review by danhumphreys-moj — APPROVED** [review]
LGTM!

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/seed/cas1/Cas1SiteSurveyPremiseFactory.kt (hunk 3 above)
Comment (davidatkinsuk) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/seed/cas1/Cas1SiteSurveyPremiseFactory.kt:8]: Yep, it helps split the logic up. Also this should make it easy for you to grab the qcode in that job from the first worksheet, with a little bit of refactoring

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/seed/cas1/Cas1SeedPremisesFromSiteSurveyXlsxJob.kt (hunk 2 above)
Comment (davidatkinsuk) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/seed/cas1/Cas1SeedPremisesFromSiteSurveyXlsxJob.kt:261]: I noticed that but i didn't really see the value given the seed service is just catching all exceptions anyway. I find `error("asdasd")` a more fluent way of raising unrecoverable errors.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/seed/cas1/Cas1SeedPremisesFromSiteSurveyXlsxJob.kt (hunk 1 above)
Comment (davidatkinsuk) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/seed/cas1/Cas1SeedPremisesFromSiteSurveyXlsxJob.kt:193]: Yes. If we refactored the room seed job to use an internal model before persisting (some data classes) we could then load the current entities into the same internal classes and generate a diff. This is less important in the short term as SWSC don't have any rooms defined to modify

**Pull request merged**